### PR TITLE
Support inline caption in figures

### PR DESCRIPTION
**Pull request by SilasBerger**
If a figure doesn't have a caption, it can sometimes look odd to have the figcaption bar appended below the image (e.g. when using https://github.com/SilasBerger/teaching-website/blob/main/docs/material/Components-Gallery/04-Hero-Image.mdx).

Therefore, this change introduces the `--inlineCaption` option which moves the entire caption onto the image itself, right at the bottom. This feature is mainly intended to be used without a caption (i.e. only the source ref), since the caption itself will generally be difficult to read.

### CODE REVIEW

File: docs/Komponentengalerie/images.md [hunk 1]
```diff
@@ -24,13 +24,23 @@ mit einem gleichnamigen `.json`-File:
 ```
 
 <BrowserWindow>
-![Streetfood Festival --width=450px --border-radius=20px](./images/street-food.jpg)
+![Streetfood Festival --width=450px](./images/street-food.jpg)
```
Comment (lebalz) [inline docs/Komponentengalerie/images.md:27]: good catch

File: src/components/Figure/styles.module.scss [hunk 2]
```diff
@@ -13,10 +14,22 @@
...
+
+            &:hover {
+                opacity: 100%;
+            }
+        }
     }
 }
 
-:globale .card__image {
+:global .card__image {
```
Comment (lebalz) [inline src/components/Figure/styles.module.scss:32]: there are some places where `:global <selector>` is used - these were introduced when the propper syntax was not clear to the author ;) It should be

```suggestion
:global(.card__image) {
```

File: src/plugins/remark-images/plugin.ts [hunk 3]
```diff
@@ -124,6 +124,10 @@ const plugin: Plugin = function plugin(
                     }
                 } as MdxJsxFlowElement | MdxJsxTextElement;
 
+                if ((options as any).inlineCaption) {
+                    caption.attributes.push(toJsxAttribute('className', 'inline'));
+                }
```
Comment (lebalz) [inline src/plugins/remark-images/plugin.ts:129]: I like this idea - could we add e plugin-config option to opt-out from automatically inlining empty captions (for example `preventInliningEmptyCaptions`)?
```suggestion
                const inlined = !optionsInput.preventInliningEmptyCaptions && /\s+/.test(cleanedAlt)
                if (inlined || (options as any).inlineCaption) {
                    caption.attributes.push(toJsxAttribute('className', 'inline'));
                }
```

File: src/components/Figure/styles.module.scss (hunk 2 above)
Comment (SilasBerger) [reply src/components/Figure/styles.module.scss]: That syntax has now also become clear to this author, thanks ;)

**Comment by SilasBerger** [comment]
> Cool addition - i like it and i'm wondering if this shouldn't be the default - inlining empty captions?

I like that! The new behavior now is as follows:
- The source ref is automatically inlined if the caption is empty and the plugin option `inlineEmptyCaptions` is NOT set to `false`
- Inlining can still be enforced or prevented for a particular image by using `My caption --inlineCaption={true}` or `--inlineCaption={false}` respectively.

I think this gives us the best balance of control and convenience, with sane defaults :).

**Pull request merged**